src/main/scala/com/microsoft/hyperspace/actions/OptimizeAction.scala

```diff
@@ -135,11 +135,12 @@ class OptimizeAction(
   override def logEntry: LogEntry = {
     // Update `previousIndexLogEntry` to keep `filesToIngore` files and append to it
     // the list of newly created index files.
-    val absolutePath = PathUtils.makeAbsolute(indexDataPath)
-    val newContent = Content.fromDirectory(absolutePath, fileIdTracker)
+    val absolutePath = PathUtils.makeAbsolute(indexDataPath.toString)
+    val newContent = Content.fromDirectory(absolutePath, fileIdTracker,
+      hadoopConfiguration = spark.sessionState.newHadoopConf())
```

Comment:
Shall we do the following? It's a bit confusing when to use hadoop config or not:
```scala
val hadoopConf = spark.sessionState.newHadoopConf())
val absolutePath = PathUtils.makeAbsolute(indexDataPath, hadoopConf)
val newContent = Content.fromDirectory(absolutePath, fileIdTracker, hadoopConfiguration = hadoopConf)
```

Reply:
👍